include/deal.II/base/utilities.h

```diff
@@ -1217,31 +1214,19 @@ namespace Utilities
         // use buffer as the target of a compressing
         // stream into which we serialize the current object
         const std::size_t previous_size = dest_buffer.size();
+        {
+          boost::iostreams::filtering_ostreambuf fosb;
 #ifdef DEAL_II_WITH_ZLIB
-        if (allow_compression)
-          {
-            boost::iostreams::filtering_ostream out;
-            out.push(
-              boost::iostreams::gzip_compressor(boost::iostreams::gzip_params(
-                boost::iostreams::gzip::default_compression)));
-            out.push(boost::iostreams::back_inserter(dest_buffer));
-
-            boost::archive::binary_oarchive archive(out);
-            archive << object;
-            out.flush();
-          }
-        else
+          if (allow_compression)
+            fosb.push(boost::iostreams::zlib_compressor());
```

Comment:
Any reason to switch from the `gzip_compressor` used before to the `zlib_compressor` here?

Reply:
To my knowledge, `gzip` uses the `zlib` algorithms internally and additionally provides tools for file compression. Since we are dealing with data only here, it felt more appropriate to use the `zlib` wrappers here. So in our context, that's analogous to wrapping a package in either green or red paper.

Since we specifically look for `zlib` in the cmake configuration phase and provide a `DEAL_II_WITH_ZLIB` flag, it appears even more appropriate to use the `zlib` wrappers.

Reply:
Interesting, the sizes of the compressed objects in test `base/utilities_pack_unpack_07` vary depending on the routines used. Objects are negligibly smaller with `zlib` compression. We could discuss about using `zlib` in a separate PR if you like.

Reply:
"that's analogous to wrapping a package in either green or red paper" -- what a timely Christmas-themed analogy :-)

Reply:
I don't really care much either way. @kronbichler did some tests a while ago that showed that compression can be surprisingly expensive in terms of CPU time.